frontend/server/src/DAO/ProblemsTags.php

```diff
@@ -44,6 +44,47 @@ public static function getProblemTags(
         return \OmegaUp\MySQLConnection::getInstance()->GetAll($sql, $params);
     }
 
+    /**
+     * Returns the list of tags of a problem. That list will contain
+     * either public or private tags.
+     *
+     * @return list<string>
+     */
+    public static function getTagsForProblem(
+        \OmegaUp\DAO\VO\Problems $problem,
+        bool $publicOnly,
+        bool $public = true
```

Comment:
se puede hacer esto no-opcional?

```suggestion
        bool $public
```